zcash_primitives/src/transaction/builder.rs

```diff
@@ -308,6 +308,8 @@ pub struct Builder<'a, P, U: sapling::builder::ProverProgress> {
     transparent_builder: TransparentBuilder,
     sapling_builder: Option<sapling::builder::Builder>,
     orchard_builder: Option<orchard::builder::Builder>,
+    #[cfg(all(zcash_unstable = "nu7", feature = "zip-233"))]
+    zip233_amount: Option<Zatoshis>,
```

Comment:
This field should not be optional; it should be set to zero instead of None in order to avoid there being multiple representations of the zero value.